docs/docs/tutorials/jwt_authentication.md

```diff
@@ -0,0 +1,517 @@
+
+    if (user != null && _passwords[username] == password) {
+      return user;
+    }
+
+    return null;
+  }
+}
+
+```
+
+We also need to provide our `Authenticator` to our routes. It will be used by the sign in route and all others that
+require authentication.
+
+In order to do so, we can use [Dart frog's dependency injection](https://dartfrog.vgv.dev/docs/basics/dependency-injection)
+and create a middleware in the root of our `routes` folder with the following code:
+
+```dart
+// lib/_middleware.dart
+import 'package:authenticated_app/authenticator.dart';
+import 'package:dart_frog/dart_frog.dart';
+
+Handler middleware(Handler handler) {
+  return handler.use(
+    provider<Authenticator>(
+      (_) => Authenticator(),
+    ),
+  );
+}
+```
+
+:::tip
+Install and use the [Dart Frog VS Code extension](https://marketplace.visualstudio.com/items?itemName=VeryGoodVentures.dart-frog) to easily create new middleware within your IDE.
+:::
+
+## Writing a Sign In route
+
+Now that we have all the domain code necessary to authenticate users given an username and a password, we can create a route to authenticate users. Within the routes folder, create the file below:
+
+```dart
+// routes/sign_in.dart
+import 'dart:io';
+
+import 'package:authenticated_app/authenticator.dart';
+import 'package:dart_frog/dart_frog.dart';
+
+Future<Response> onRequest(RequestContext context) async {
+  return switch (context.request.method) {
+    HttpMethod.post => _onPost(context),
+    _ => Future.value(
+        Response(statusCode: HttpStatus.methodNotAllowed),
+      ),
+  };
+}
+
+Future<Response> _onPost(RequestContext context) async {
+  final body = await context.request.json() as Map<String, dynamic>;
+  final username = body['username'] as String?;
+  final password = body['password'] as String?;
+
+  if (username == null || password == null) {
+    return Response(statusCode: HttpStatus.badRequest);
+  }
+
+  final authenticator = context.read<Authenticator>();
+
+  final user = authenticator.findByUsernameAndPassword(
+    username: username,
+    password: password,
+  );
+
+  if (user == null) {
+    return Response(statusCode: HttpStatus.unauthorized);
+  } else {
+    return Response.json(
+      body: { 'token': username },
+    );
+  }
+}
+```
+
+:::tip
+Install and use the [Dart Frog VS Code extension](https://marketplace.visualstudio.com/items?itemName=VeryGoodVentures.dart-frog) to easily create new routes within your IDE.
+:::
+
+To people familiar with Dart Frog, the code above should be no real challenge, we are simply handling the
+request in the following steps:
+
+- Check if we have all the info needed, returning `badRequest` otherwise.
+- Get our `Authenticator` dependency from our dependency injection.
+- Use the `Authenticator` to get a user that match the request's credential.
+- Returns `unauthorized` (401) if there is no user, or returns the user username as the authentication token otherwise.
```

Comment:
```suggestion
- Return an `unauthorized` (401) if there is no user, or return the user username as the authentication token otherwise.
```

Present tense?